### PR TITLE
Make tags modifiable

### LINKED ISSUE #10: Modify Tag List
#### As a user I want to edit the tag list for an image.

__Story__:

Image tags are awesome -- our user's love them! Tags help us make sense of the
plethora of images added to our system. But how annoying is it when our users
make spelling mistakes in their image tags? Super annoying! How awesome, and
useful would it be to allow for tags to be edited after an image is added?
Super awesome, and useful!

__Acceptance criteria__:
- [x] There is a link to edit the tags for an image.
- [x] The image whose tags are to be modified is displayed on the edit page.
- [x] On the edit page, I can change the tags to any valid tag list.
- [x] After saving the edit page with valid tags, my changes are visible
  throughout the rest of the application.

__Discussion__:
- [x] Should an image's URL be editable?
- [x] How would you define a "valid" tag list? 

__Dependencies__:
- Image Tags
- Refactor flow tests to use page objects

### PR DESCRIPTION
**Pull request by jasonmally**
Closes #10

### CODE REVIEW

**Comment by jasonmally** [comment]
TODO: add controller tests

File: app/controllers/image_controller.rb [hunk 1]
```diff
@@ -30,6 +30,21 @@ def show
     @image = Image.find(valid_image_shown)
   end
 
+  def edit
+    @image = Image.find(valid_image_shown)
+    render 'update'
```
Comment (xuinleaf849) [inline app/controllers/image_controller.rb]: just FYI, conventionally it's edit who do the render job:)

File: db/migrate/20190703171139_add_tags_if_none.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    execute <<~SQL
+      DELETE FROM taggings WHERE tag_id = '#{placeholder_id}'
+    SQL
+
+    execute <<~SQL
+      DELETE FROM tags WHERE name = '#{PLACEHOLDER_TAG}'
+    SQL
+  end
+end
```
Comment (xuinleaf849) [inline db/migrate/20190703171139_add_tags_if_none.rb]: good job!

File: app/models/image.rb [hunk 3]
```diff
@@ -2,5 +2,6 @@ class Image < ApplicationRecord
   validates :name, presence: { message: 'Please enter a title for your image.' }
   validates :image_url, presence: { message: 'Please enter a URL for your image.' }
   validates :image_url, url: { message: 'This image URL is invalid.' }
+  validates :tag_list, presence: { message: 'You cannot submit an image with no tags!' }
```
Comment (xuinleaf849) [inline app/models/image.rb]: You don't have to do it, but would be better if there's a test for this in `image_test.rb`

**Pull request merged**